File: Backend/HomeApplianceStore.Domain/Entities/Order.cs
using System;

namespace HomeApplianceStore.Domain.Entities;

public class Order
{
    public int OrderId { get; set; }
    public int? UserId { get; set; }
    public int? VoucherId { get; set; }
    public DateTime? OrderDate { get; set; }
    public string ShippingName { get; set; } = null!;
    public string ShippingPhone { get; set; } = null!;
    public string ShippingAddress { get; set; } = null!;
    public decimal TotalAmount { get; set; }
    public decimal? DiscountAmount { get; set; }
    public decimal FinalAmount { get; set; }
    public byte PaymentMethod { get; set; }
    public byte PaymentStatus { get; set; }
    public byte OrderStatus { get; set; }
    public string? Note { get; set; }
}

File: Frontend/src/types/order.ts
export interface CheckoutRequest {
    shippingName: string;
    shippingPhone: string;
    shippingAddress: string;
    paymentMethod: number; // 1: COD, 2: Bank, 3: VNPay, 4: Momo
    note?: string;
    voucherCode?: string;
}

export interface OrderHistory {
    orderId: number;
    orderDate: string;
    totalAmount: number;
    finalAmount: number;
    orderStatus: number;
    paymentMethod: number;
    paymentStatus: number;
    firstProductTitle: string;
    firstProductImageUrl: string;
    totalProductCount: number;
}

export interface OrderDetailItem {
    productId: number;
    productName: string;
    imageUrl: string;
    quantity: number;
    unitPrice: number;
    totalPrice: number;
}

export interface OrderDto {
    orderId: number;
    orderDate: string;
    orderStatus: number;
    paymentMethod: number;
    paymentStatus: number;
    shippingName: string;
    shippingPhone: string;
    shippingAddress: string;
    note?: string;
    totalAmount: number;
    discountAmount?: number;
    finalAmount: number;
    items: OrderDetailItem[];
}
